initramfs/src/image.rs

```diff
@@ -175,3 +195,115 @@ impl Image {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::{FileHandler, Image};
+    use anyhow::Ok;
+    use std::env;
+    use std::io::{Read, Write};
+
+    const VALID_IMAGE_NAME: &str = "my_awesome_img:14md35";
+    const VALID_IMAGE_NAME_FROM_HUB: &str = "bitnami/mongodb:latest";
+
+    #[derive(Debug, Clone)]
+    struct MockFileHandler {
+        path: String,
+    }
+
+    impl Read for MockFileHandler {
+        fn read(&mut self, _buf: &mut [u8]) -> std::io::Result<usize> {
+                std::result::Result::Ok(0)
+        }
+
+        fn read_to_end(&mut self, buf: &mut Vec<u8>) -> std::io::Result<usize> {
+            std::result::Result::Ok(buf.len())
+        }
+    }
+
+    impl FileHandler for MockFileHandler {
+        fn create(path: &str) -> anyhow::Result<Self>
+        where
+            Self: std::marker::Sized,
+        {
+            Ok(Self {
+                path: path.to_string(),
+            })
+        }
+
+        fn open(path: &str) -> anyhow::Result<Self>
+        where
+            Self: std::marker::Sized,
+        {
+            MockFileHandler::create(path)
```

Comment:
There could be some problems if someone needs to change create() behaviour 

```suggestion
            Ok(Self {
                path: path.to_string(),
            })
```